src/examples/java/com/amazonaws/crypto/examples/keyring/rawrsa/PublicPrivateKeySeparate.java

```diff
@@ -0,0 +1,144 @@
+        // https://www.nist.gov/publications/transitioning-use-cryptographic-algorithms-and-key-lengths
+        kg.initialize(4096);
+        final KeyPair keyPair = kg.generateKeyPair();
+
+        // Create the keyring that determines how your data keys are protected.
+        //
+        // Create the encrypt keyring that only has access to the public key.
+        final Keyring publicKeyKeyring = StandardKeyrings.rawRsaBuilder()
+                // The key namespace and key name are defined by you
+                // and are used by the raw RSA keyring
+                // to determine whether it should attempt to decrypt
+                // an encrypted data key.
+                //
+                // https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/choose-keyring.html#use-raw-rsa-keyring
+                .keyNamespace("some managed raw keys")
+                .keyName("my RSA wrapping key")
+                .publicKey(keyPair.getPublic())
+                // The padding scheme tells the raw RSA keyring
+                // how to use your wrapping key to encrypt data keys.
+                //
+                // We recommend using OAEP_SHA256_MGF1.
+                // You should not use PKCS1 unless you require it for backwards compatibility.
+                .paddingScheme(RawRsaKeyringBuilder.RsaPaddingScheme.OAEP_SHA256_MGF1)
+                .build();
+
+        // Create the decrypt keyring that has access to the private key.
+        final Keyring privateKeyKeyring = StandardKeyrings.rawRsaBuilder()
+                // The key namespace and key name MUST match the encrypt keyring.
+                .keyNamespace("some managed raw keys")
+                .keyName("my RSA wrapping key")
+                .privateKey(keyPair.getPrivate())
+                // The padding scheme MUST match the encrypt keyring.
+                .paddingScheme(RawRsaKeyringBuilder.RsaPaddingScheme.OAEP_SHA256_MGF1)
+                .build();
+
+        // Encrypt your plaintext data using the encrypt keyring.
+        final AwsCryptoResult<byte[]> encryptResult = awsEncryptionSdk.encrypt(
+                EncryptRequest.builder()
+                        .keyring(publicKeyKeyring)
+                        .encryptionContext(encryptionContext)
+                        .plaintext(sourcePlaintext).build());
+        final byte[] ciphertext = encryptResult.getResult();
+
+        // Demonstrate that the ciphertext and plaintext are different.
+        assert !Arrays.equals(ciphertext, sourcePlaintext);
+
+        // Try to decrypt your encrypted data using the *encrypt* keyring.
+        // This demonstrates that you cannot decrypt using the public key.
+        try {
+            awsEncryptionSdk.decrypt(
+                    DecryptRequest.builder()
+                            .keyring(publicKeyKeyring)
+                            .ciphertext(ciphertext)
+                            .build());
+            assert false;
```

Comment:
We should provide an explanation for what we are trying to assert here.

IIRC, this would be:
```suggestion
            throw new AssertionError("The public key can never decrypt!");
```

Reply:
Yep that works